agent/api_manager/osv_service_api.py

```diff
@@ -0,0 +1,194 @@
+            vuln = VulnData(
+                risk=risk,
+                description=description,
+                summary=summary,
+                fixed_version=fixed_version,
+                cvss_v3_vector=cvss_v3_vector,
+                references=vulnerability.get("references"),
+                cves=filtered_cves,
+            )
+            parsed_vulns.append(vuln)
+
+        return parsed_vulns
+
+    except json.JSONDecodeError as e:
+        logger.error(f"Error decoding JSON: {e}")
+        return []
+
+
+def construct_vuln(
+    parsed_vulns: list[VulnData], package_name: str | None, package_version: str | None
+) -> Iterator[osv_output_handler.Vulnerability]:
+    """Construct Vulneravilities from the parse output.
+    Args:
+        parsed_vulns: list of VulnData.
+        package_name: The package name.
+        package_version: The package version.
+    Yields:
+        Vulnerability entry.
+    """
+    for vuln in parsed_vulns:
+        description = (
+            f"Dependency `{package_name}` with version `{package_version}`"
+            f"has a security issue.\nThe issue is identified by CVEs: `{', '.join(vuln.cves)}`."
+        )
+        recommendation = (
+            f"We recommend updating `{package_name}` to the latest available version."
+        )
+        yield osv_output_handler.Vulnerability(
+            entry=kb.Entry(
+                title=f"Use of Outdated Vulnerable Component: "
+                f"{package_name}@{package_version}: {', '.join(vuln.cves)}",
+                risk_rating=vuln.risk,
+                short_description=vuln.summary,
+                description=description,
+                references=osv_output_handler.build_references(vuln.references),
+                security_issue=True,
+                privacy_issue=False,
+                has_public_exploit=False,
+                targeted_by_malware=False,
+                targeted_by_ransomware=False,
+                targeted_by_nation_state=False,
+                recommendation=recommendation,
+            ),
+            technical_detail=f"{vuln.description} \n#### CVEs:\n {', '.join(vuln.cves)}",
+            risk_rating=RISK_RATING_MAPPING[vuln.risk],
+        )
+
+
+def _get_fixed_version(
+    affected_data: list[dict[str, list[dict[str, list[dict[str, str]]]]]],
+) -> str:
+    fixed_version = ""
+    if affected_data is not None:
+        try:
+            ranges_data: list[dict[str, list[dict[str, str]]]] = affected_data[0].get(
+                "ranges", []
+            )
+            if ranges_data:
+                events_data = ranges_data[0].get("events", [])
+                if len(events_data) > 1:
+                    fixed_version = events_data[1].get("fixed", "")
+        except IndexError:
+            logger.warning("Can't get the fixed version.")
+
+    return fixed_version
+
+
+def _get_cvss_v3_vector(severity_data: list[dict[str, str]]) -> str:
+    cvss_v3_vector = ""
+    if severity_data:
```

Comment:
explicit? with None and count and you don't have to catch `IndexError` just avoid it before.